Home5/Home5/BestMark.cs

```diff
@@ -0,0 +1,162 @@
+        }
+
+        /// <summary>
+        /// This method is searching for best Physical Education mark in Group
+        /// </summary>
+        public void PhysicalEducationBestMark(Student[] arrayOfStudents)
+        {
+            int maxPhysicalEducationMark = 0;
+            string name = "";
+
+            for (int i = 0; i < arrayOfStudents.Length; i++)
+            {
+                int mark = arrayOfStudents[i].PhysicalEducationMark;
+
+                if (mark > maxPhysicalEducationMark)
+                {
+                    maxPhysicalEducationMark = mark;
+                    name = arrayOfStudents[i].Name;
+                }
+            }
+
+            Console.WriteLine($"Name: {name}, Physical Education mark: {maxPhysicalEducationMark}");
+        }
+
+        /// <summary>
+        /// This method is searching for best Biology mark in Group
+        /// </summary>
+        public void BiologyBestMark(Student[] arrayOfStudents)
+        {
+            int maxBiologyMark = 0;
+            string name = "";
+
+            for (int i = 0; i < arrayOfStudents.Length; i++)
+            {
+                int mark = arrayOfStudents[i].BiologyMark;
+
+                if (mark > maxBiologyMark)
+                {
+                    maxBiologyMark = mark;
+                    name = arrayOfStudents[i].Name;
+                }
+            }
+
+            Console.WriteLine($"Name: {name}, Biology mark: {maxBiologyMark}");
+        }
+
+        /// <summary>
+        /// This method calculates Average Math mark in Group
+        /// </summary>
+        public void MathAverageMark(Student[] arrayOfStudents)
+        {
+            double mathAverageMark = 0;
+            double markSum = 0;
+            string group = arrayOfStudents[0].Group;
+
+            for (int i = 0; i < arrayOfStudents.Length; i++)
+            {
+                markSum += arrayOfStudents[i].MathMark;
+            }
+
+            mathAverageMark = markSum / arrayOfStudents.Length;
+
+            Console.WriteLine($"{group}, Math Average mark: {mathAverageMark}");
+        }
+
+        /// <summary>
+        /// This method calculates Average Physical Education mark in Group
+        /// </summary>
+        public void PhysicalEducationAverageMark(Student[] arrayOfStudents)
+        {
+            double pEAverageMark = 0;
+            double markSum = 0;
+            string group = arrayOfStudents[0].Group;
+
+            for (int i = 0; i < arrayOfStudents.Length; i++)
+            {
+                markSum += arrayOfStudents[i].PhysicalEducationMark;
+            }
+
+            pEAverageMark = markSum / arrayOfStudents.Length;
+
+            Console.WriteLine($"{group}, Physical Education Average mark: {pEAverageMark}");
+        }
+
+        /// <summary>
+        /// This method calculates Average Biology mark in Group
+        /// </summary>
+        public void BiologyAverageMark(Student[] arrayOfStudents)
```

Comment:
It is better to use a separate class for that. 
The class is called BestMark, but we count average here.